modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeFactory.java

```diff
@@ -69,30 +72,27 @@ public class IgniteTypeFactory extends JavaTypeFactoryImpl {
     private static final SqlIntervalQualifier INTERVAL_QUALIFIER_DAY_TIME = new SqlIntervalQualifier(TimeUnit.DAY,
             TimeUnit.SECOND, SqlParserPos.ZERO);
 
+    public static final IgniteTypeFactory INSTANCE = new IgniteTypeFactory(IgniteTypeSystem.INSTANCE);
+
     /** Default charset. */
     private final Charset charset;
 
     /** A registry that contains custom data types. **/
     private final CustomDataTypes customDataTypes;
 
-    /** Contains java types internally mapped into appropriate rel types. */
-    private static final Map<Class<?>, Supplier<RelDataType>> implementedJavaTypes = new IdentityHashMap<>();
-
     {
```

Comment:
we have static implementedJavaTypes, but filing it for each instances - let's remove `static` modification for implementedJavaTypes

Reply:
good catch! fixed